packages/ui/components/ui/tx/view/swap.tsx

```diff
@@ -1,56 +1,51 @@
 import { ViewBox } from './viewbox';
 import { SwapView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/dex/v1/dex_pb';
-import { bech32Address } from '@penumbra-zone/types/src/address';
 import { fromBaseUnitAmount, joinLoHiAmount } from '@penumbra-zone/types/src/amount';
 import { uint8ArrayToBase64 } from '@penumbra-zone/types/src/base64';
+import { ActionDetails } from './action-details';
+import { AddressView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
+import { AddressViewComponent } from './address-view';
 
 export const SwapViewComponent = ({ value }: { value: SwapView }) => {
   if (value.swapView.case === 'visible') {
     const { tradingPair, delta1I, delta2I, claimFee, claimAddress } =
       value.swapView.value.swapPlaintext!;
 
-    const encodedAddress = bech32Address(claimAddress!);
+    const addressView = new AddressView({
+      addressView: { case: 'decoded', value: { address: claimAddress } },
+    });
 
     return (
       <ViewBox
         label='Swap'
         visibleContent={
-          <div className='flex flex-col gap-2'>
-            <div>
-              <b>Asset 1:</b>
-              <div className='ml-5'>
-                <b>ID: </b>
+          <div className='flex flex-col gap-8'>
```

Comment:
Before:
<img width="760" alt="image" src="https://github.com/penumbra-zone/web/assets/1121544/80bde8ab-e2d6-4c3d-8066-326d13393601">

After:
<img width="762" alt="image" src="https://github.com/penumbra-zone/web/assets/1121544/1a668f95-a816-40ef-8eeb-03cb59c25ad0">

Reply:
(In follow-up PR(s), this will move closer to [this proposed design](https://github.com/penumbra-zone/web/issues/424#issuecomment-1947375109).)